### DISCUSSION TITLE
生成目标如何放在子文件夹中？

### DISCUSSION THREAD

**Discussion by Extra-Creativity** [xmake-io/xmake#3294] [Q&A]
比如结构如下：
```
├── test1
│   ├── file1.cpp
│   ├── file1.h
│   └── xmake.lua
└── xmake.lua
```

我想让外层的xmake.lua的`target`在`build/linux/x86_64/release/`下，而让内层的`target`在`build/linux/x86_64/release/test1/`下，依此类推，有什么直接的方法吗？

**Comment by star-hengxing** [comment]
https://xmake.io/#/zh-cn/manual/project_target?id=targetset_targetdir

获取当前目录拼接一下路径，然后用这个 api？

**Reply by Extra-Creativity** [reply]
但是这样VSCode的xmake插件不会变成文件夹形式的，比如CMake的话是这样：
![image](https://user-images.githubusercontent.com/72960952/212536539-13a72a78-2f12-4d0d-a5e9-e0eabb7a6583.png)
但是XMake改了还是所有生成的target都在一起，像这样：
![image](https://user-images.githubusercontent.com/72960952/212536676-afc91af3-ae9f-4807-ba57-299799f42c6d.png)

而且每个target都写感觉有点麻烦？

**Reply by waruqi** [reply]
全局设置

```lua
on_load(function (target)
    target:set("targetdir", "$(buildir)/$(plat)/$(arch)/$(mode)/" .. target:name())
end)
```

**Reply by Extra-Creativity** [reply]
简单重写了一下，满足我的生成要求了。
```lua
on_load(function (target)
    local projectDir = os.projectdir()
    local scriptDir = target:scriptdir()
    -- +2 because of 1-indexed and an additional '/'
    local relativeDir = string.sub(scriptDir, #projectDir + 2)
    target:set("targetdir", path.join("$(buildir)", "$(plat)", "$(arch)", "$(mode)", relativeDir))
end)
```

BTW，如果可以的话希望能在这里改进一下VSCode的插件（如果不能那就这样吧:-) ）。

**Reply by waruqi** [reply]
set_group

**Reply by Extra-Creativity** [reply]
OK了，加一个`target:set("group", relativeDir)`就一样了，十分感谢！
